tensornetwork/contractors/custom_path_solvers/nconinterface.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import numpy as np
+from typing import List, Union, Any, Tuple, Type, Optional, Sequence
+
+from tensornetwork.contractors.custom_path_solvers.pathsolvers import full_solve_complete
+
+
+#------------------------------------------------------------------------#
+def ncon_solver(tensors: List[np.ndarray],
+                connects: List[List[int]],
```

Comment:
`connects` -> `network_structure` (to be consistent with `ncon`)

Reply:
Is there a better name that we could give this? I am not a big fan of `network_structure` as a variable name as it both rather long (especially when we have to add prefixes like `flat_network_structure` ) as well as vague/misleading (since a networks structure usually refers to some overall characteristic such as 'centralized' or 'distributed'). Some alternatives could be `labels` or `ind_labels` or `edge_labels` or `net_labels`. Even `connects` is more informative in my opinion, as it hints to being the set of network connections.

Reply:
`labels` would be my favorite, let's do those (we only need to keep argument names consistent, names inside routines don't matter so much)

Reply:
sure `labels` is great, I will change the arguments to that.